Xamarin.AndroidBinderator/Engine.cs

```diff
@@ -150,6 +152,21 @@ static async Task DownloadArtifacts(MavenRepository maven, BindingConfig config,
 				using (var astrm = await mvnArt.OpenLibraryFile(mavenArtifact.Version, mavenProject.Packaging))
 				using (var sw = File.Create(artifactFile))
 					await astrm.CopyToAsync(sw);
+
+				// Determine MD5
+				try
+				{
+					// First try download
+					using (var astrm = await mvnArt.OpenLibraryFile(mavenArtifact.Version, mavenProject.Packaging + ".md5"))
+					using (var sw = File.Create(md5File))
+						await astrm.CopyToAsync(sw);
+				}
+				catch
+				{
+					// Then hash the downloaded artifact
+					using (var file = File.OpenRead(artifactFile))
+						File.WriteAllText(md5File, HashMd5(file));

```

Comment:
Not sure how we feel about this, but if something goes wrong, we just generate the MD5 from the artifact that we just downloaded.